`import-x/no-unresolved` cannot resolve `@crxjs/vite-plugin` in `vite.config.ts`

### Build tool

Vite

### Where do you see the problem?

- [ ] In the browser
- [x] In the terminal

### Describe the bug

After upgrading `@crxjs/vite-plugin` from `2.3.0` to `2.4.0`, ESLint reports `@crxjs/vite-plugin` as unresolved in `vite.config.ts`, even though Vite, TypeScript, and the build still work.

In this case the failing rule is [`import-x/no-unresolved`](https://github.com/un-ts/eslint-plugin-import-x/blob/master/docs/rules/no-unresolved.md) (from [`eslint-plugin-import-x`](https://github.com/un-ts/eslint-plugin-import-x)).

```sh
/home/.../vite.config.ts
  1:21  error  Unable to resolve path to module '@crxjs/vite-plugin'  import-x/no-unresolved
```

This appears to be the same class of resolver problem as my earlier `eslint-plugin-import` issue:

- #684 

### Reproduction

1. Clone https://github.com/karlhorky/archive-webpage-browser-extension
2. Check out the commit  https://github.com/karlhorky/archive-webpage-browser-extension/commit/8ee512982e455d9ba6dbcb2b292a479c0abb1163 with detached head
3. Install dependencies
4. Run `pnpm eslint . --max-warnings 0`
5. Observe the error message below

```bash
./vite.config.ts
  1:21  error  Unable to resolve path to module '@crxjs/vite-plugin'  import-x/no-unresolved

✖ 1 problem (1 error, 0 warnings)
```

### Logs

```shell

```

### System Info

```shell
System:
  OS: Linux
  Binaries:
    Node: 24.14.1
    npm: 11.11.0
    pnpm: 10.33.0
  npmPackages:
    @crxjs/vite-plugin: 2.4.0
    vite: 7.3.2
    eslint: 10.2.1
```

### Severity

blocking an upgrade

--- a/.changeset/vite-plugin-esm-declarations.md
+++ b/.changeset/vite-plugin-esm-declarations.md
@@ -1,0 +1,5 @@
+---
+"@crxjs/vite-plugin": patch
+---
+
+fix: publish ESM declarations for the Vite plugin entrypoint

--- a/.github/workflows/vite-plugin.yml
+++ b/.github/workflows/vite-plugin.yml
@@ -50,6 +50,9 @@ jobs:
       - name: Lint
         run: pnpm lint
 
+      - name: Check package type exports
+        run: pnpm lint:attw
+
   unit-output:
     name: Unit & Output Tests (${{ matrix.os }})
     runs-on: ${{ matrix.os }}

--- a/packages/vite-plugin/package.json
+++ b/packages/vite-plugin/package.json
@@ -28,7 +28,7 @@
   "exports": {
     ".": {
       "import": {
-        "types": "./dist/index.d.ts",
+        "types": "./dist/index.d.mts",
         "default": "./dist/index.mjs"
       },
       "require": {
@@ -62,6 +62,8 @@
     "dev:js": "npm run build:js -- -w",
     "dev:lint": "tsc --noEmit --watch",
     "lint": "run-s lint:eslint lint:types",
+    "lint:attw": "run-s build lint:attw:check",
+    "lint:attw:check": "attw --pack --entrypoints . .",
     "lint:eslint": "eslint \"{src,test}/**/*.ts\"",
     "lint:types": "tsc --noEmit",
     "test": "vitest --mode unit",
@@ -91,6 +93,7 @@
     "rxjs": "7.5.7"
   },
   "devDependencies": {
+    "@arethetypeswrong/cli": "0.18.3",
     "@extend-chrome/messages": "1.2.2",
     "@extend-chrome/storage": "1.5.0",
     "@rollup/plugin-alias": "4.0.4",

--- a/packages/vite-plugin/rollup.config.ts
+++ b/packages/vite-plugin/rollup.config.ts
@@ -119,6 +119,11 @@ const config = defineConfig([
     output: { file: 'dist/index.d.ts', format: 'es' },
     plugins: [dts()],
   },
+  {
+    input: 'src/node/index.ts',
+    output: { file: 'dist/index.d.mts', format: 'es' },
+    plugins: [dts()],
+  },
 ])
 
 export default config

--- a/packages/vite-plugin/src/node/packageExports.test.ts
+++ b/packages/vite-plugin/src/node/packageExports.test.ts
@@ -1,0 +1,36 @@
+import fs from 'fs-extra'
+import path from 'pathe'
+import { fileURLToPath } from 'url'
+import { describe, expect, test } from 'vitest'
+
+const __filename = fileURLToPath(import.meta.url)
+const __dirname = path.dirname(__filename)
+const packageRoot = path.resolve(__dirname, '../..')
+
+describe('package exports', () => {
+  test('uses ESM declarations for the ESM entrypoint', async () => {
+    const packageJson = await fs.readJson(path.join(packageRoot, 'package.json'))
+
+    expect(packageJson.exports['.']).toEqual({
+      import: {
+        types: './dist/index.d.mts',
+        default: './dist/index.mjs',
+      },
+      require: {
+        types: './index.d.cts',
+        default: './index.cjs',
+      },
+    })
+    expect(packageJson.types).toBe('dist/index.d.ts')
+  })
+
+  test('builds both legacy and ESM declaration files', async () => {
+    const rollupConfig = await fs.readFile(
+      path.join(packageRoot, 'rollup.config.ts'),
+      'utf8',
+    )
+
+    expect(rollupConfig).toContain("file: 'dist/index.d.ts'")
+    expect(rollupConfig).toContain("file: 'dist/index.d.mts'")
+  })
+})

